diff --git a/include/nanobind/nanobind.h b/include/nanobind/nanobind.h
--- a/include/nanobind/nanobind.h
+++ b/include/nanobind/nanobind.h
@@ -26,6 +26,9 @@
 #define NB_VERSION_PATCH 0
 #define NB_VERSION_DEV   0 // A value > 0 indicates a development release
 
+// Include Python.h before standard headers
+#include "nb_python.h"
+
 // Core C++ headers that nanobind depends on
 #include <cstddef>
 #include <cstdint>
@@ -39,7 +42,6 @@
 
 // Implementation. The nb_*.h files should only be included through nanobind.h
 // IWYU pragma: begin_exports
-#include "nb_python.h"
 #include "nb_defs.h"
 #include "nb_enums.h"
 #include "nb_traits.h"
